cmd/opm/render/cmd.go

```diff
@@ -100,3 +127,36 @@ func nullLogger() *logrus.Entry {
 	logger.SetOutput(io.Discard)
 	return logrus.NewEntry(logger)
 }
+
+func filtererFromFile(filterFile string, log *logrus.Entry) (filter.Filterer, error) {
+	if filterFile == "" {
+		return nil, nil
+	}
+
+	data, err := os.ReadFile(filterFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var obj struct {
+		metav1.TypeMeta `json:",inline"`
+	}
+	if err := yaml.Unmarshal(data, &obj); err != nil {
+		return nil, err
+	}
+
+	if obj.Kind != "FilterConfiguration" {
+		return nil, fmt.Errorf("unexpected kind %q", obj.Kind)
+	}
+
+	switch obj.APIVersion {
+	case "olm.operatorframework.io/v1alpha1":
+		cfg, err := configv1alpha1.LoadFilterConfiguration(data)
+		if err != nil {
+			return nil, err
+		}
+		return configv1alpha1.NewFilterer(cfg, configv1alpha1.WithLogger(log)), nil
+	default:
+		return nil, fmt.Errorf("unexpected API version %q", obj.APIVersion)
+	}
```

Comment:
It looks like the `configv1alpha1.LoadFilterConfiguration` function already performs validation. Could this whole thing just be simplified to something like:
```go
	if filterFile == "" {
		return nil, nil
	}

	data, err := os.ReadFile(filterFile)
	if err != nil {
		return nil, err
	}
	
	cfg, err := configv1alpha1.LoadFilterConfiguration(data)
	if err != nil {
		return nil, err
	}
	return configv1alpha1.NewFilterer(cfg, configv1alpha1.WithLogger(log)), nil
```
?

Reply:
I do realize that maybe this isn't as extensible because it only validates against the specific config version but this could be updated when/if we add another version

Reply:
+1, this is premature optimization. I'll reduce and add some comments about what we might do if we add new config file versions.